skimage/filters/_unsharp_mask.py

```diff
@@ -0,0 +1,113 @@
+import numpy as np
+from scipy.ndimage.filters import gaussian_filter
+from skimage.util.dtype import dtype_limits
+
+
+def _unsharp_mask_single_channel(image, radius, amount):
+    """Single channel implementation of the unsharp masking filter."""
+    blurred = gaussian_filter(image.astype(np.float_),
+                              sigma=radius,
+                              mode='reflect')
+    result = image + (image - blurred) * amount
+
+    # Enforce image dtype limits
+    if image.dtype.kind == 'f':
+        _min, _max = dtype_limits(
+            image, clip_negative=np.all(image >= 0))
+    else:
+        _min, _max = dtype_limits(image, clip_negative=False)
+    result = np.clip(result, _min, _max).astype(image.dtype)
+    return result
+
+
+def unsharp_mask(image, radius=1.0, amount=1.0, multichannel=False):
+    """Unsharp masking filter.
+
+    The sharp details are identified as the difference between the original
+    image and its blurred version. These details are then scaled, and added
+    back to the original image.
+
+    Parameters
+    ----------
+    image : (M[, N[, ..., P]][, C]) ndarray
+        Input image.
+    radius : scalar or sequence of scalars, optional
+        If a scalar is given, then its value is used for all dimensions.
+        If sequence is given, then there must be exactly one radius
+        for each dimension except the last dimension for multichannel images.
+        Note that 0 radius means no blurring, and negative values are
+        not allowed.
+    amount : scalar, optional
+        The details will be amplified with this factor. The factor could be 0
+        or negative. Typically, it is a small positive number, e.g. 1.0.
+    multichannel : bool, optional
+        If True, the last `image` dimension is considered as a color channel,
+        otherwise as spatial. Color channels are processed individually.
+
+    Returns
+    -------
+    output : (M[, N[, ..., P]][, C]) ndarray
+        Image with unsharp mask applied. It has the same shape and dtype
+        as `image`. If `image` dtype was integer, and it cannot represent
```

Comment:
Why this special clipping behavior for integers, but not floats?

Reply:
:+1: to @stefanv 's comment

Reply:
The input could be an uint8 type, e.g. with maximum value of 253.
The processing must happen in float format.
After the filter I get a float image which can be transformed back to the original type.
(Actually, this is the most common way.)

So in float, individual pixels could reach larger than 255 values. 
I like to use explicit clipping and not trusting overflow/underflow defaults in different languages, python included.

And the related question: what is the expected behavior in skimage? (See my other comment.) I did not understand it fully yet.